components/gitpod-cli/cmd/tasks.go

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/fatih/color"
+	"github.com/gitpod-io/gitpod/supervisor/api"
+	"github.com/rodaine/table"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	"golang.org/x/term"
+	"google.golang.org/grpc"
+)
+
+// tasksCmd represents the tasks command
+var tasksCmd = &cobra.Command{
+	Use:   "tasks",
+	Short: "Interact with workspace tasks",
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) == 0 {
+			cmd.Help()
+		}
+	},
+}
+
+var attachTaskCmdOpts struct {
+	Interactive bool
+	ForceResize bool
+}
+
+// TODO(andreafalzetti): refactor tasks.go to only keep the tasks cmd initialisation, moving the subcommands in their own file
+
+// TODO(andreafalzetti): move it somewhere else so that it's reusable from other cmds
+func dialSupervisor() *grpc.ClientConn {
+	supervisorAddr := os.Getenv("SUPERVISOR_ADDR")
+	if supervisorAddr == "" {
+		supervisorAddr = "localhost:22999"
+	}
+	supervisorConn, err := grpc.Dial(supervisorAddr, grpc.WithInsecure())
+	if err != nil {
+		log.WithError(err).Fatal("cannot connect to supervisor")
+	}
+
+	return supervisorConn
+}
+
+// listTasksCmd represents the tasks list command
+var listTasksCmd = &cobra.Command{
+	Use:   "list",
+	Short: "Lists the workspace tasks and their state",
+	Run: func(cmd *cobra.Command, args []string) {
+		client := api.NewStatusServiceClient(dialSupervisor())
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		// TODO(andreafalzetti): ask how to opt-out from the stream! {Observe: false} gives me a stream 😢
+		listen, err := client.TasksStatus(ctx, &api.TasksStatusRequest{Observe: false})
+		if err != nil {
+			log.WithError(err).Error("Cannot list tasks")
+		}
+
+		errchan := make(chan error, 5)
+		func() {
+			for {
+				resp, err := listen.Recv()
+				if err != nil {
+					errchan <- err
+				}
+
+				tasks := resp.GetTasks()
+
+				if tasks != nil {
+					headerFmt := color.New(color.FgGreen, color.Underline).SprintfFunc()
+					columnFmt := color.New(color.FgYellow).SprintfFunc()
+
+					tbl := table.New("ID", "Name", "State")
```

Comment:
I've tried a couple of table utils:

1. https://github.com/rodaine/table
2. https://github.com/olekukonko/tablewriter

`table` looks nice and clean, but I would suggest to continue with `tablewriter` as it allows speciyinf colors dynamically for each line while in `table` you can only set a color to a whole column. Also, `tablewriter` seems to have more activity/starts on the repo.

![Screenshot 2022-01-17 at 17 03 26](https://user-images.githubusercontent.com/2318450/149841745-4153f68f-620a-4b47-83ff-988808a663f5.png)

Now, I am not sure if this is something inline with Gitpod's preference/style (e.g. do we want to remove colors? do we want to use a custom tab-based output instead of a table?)

I quite like the table as it's nice and tidy, and the colours add a nice touch.